src/core/src/factory.rs

```diff
@@ -427,7 +427,7 @@ pub trait Factory<R: Resources> {
         self.view_texture_as_depth_stencil(tex, 0, None, texture::DepthStencilFlags::empty())
     }
 	
-	fn create_texture_immutable_u8_mipmapped<T: format::TextureFormat>(&mut self, kind: texture::Kind, data: &[&[u8]], mipmap: Option<texture::Mipmap>)
+	fn create_texture_immutable_u8_mipmapped<T: format::TextureFormat>(&mut self, kind: texture::Kind, mipmap: texture::Mipmap, data: &[&[u8]])
```

Comment:
given that we are changing the API anyway in a breaking fashion, I don't think it makes sense to try to preserve the old `create_texture_immutable_u8`. Let's just add a new parameter in there.

Reply:
I'd also suggest either naming the `Mipmap` options more self-describing (`ProvidedWithBitmapData`, `AllocatedForGeneration`) or document it better to eliminate usage `Allocated` without further generation call.